src/WebJobs.Script.WebHost/Middleware/ResponseContextItemsCheckMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    internal class ResponseContextItemsCheckMiddleware
+    {
+        private readonly ILogger _logger;
+        private readonly RequestDelegate _next;
+
+        public ResponseContextItemsCheckMiddleware(RequestDelegate next, ILogger<ExceptionMiddleware> logger)
+        {
+            _logger = logger;
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (_next != null)
+            {
+                await _next(context);
+            }
+
+            // Check response for duplicate http headers
+            object value = null;
+            if (context?.Items?.TryGetValue(ScriptConstants.AzureFunctionsDuplicateHttpHeaderKey, out value) ?? false)
+            {
+                _logger.LogInformation($"Duplicate HTTP header from function invocation removed. Key: '{value.ToString() ?? string.Empty}'.");
```

Comment:
Should we log this as Debug? If someone has App Insights set up (with an out-of-proc language) and call a downstream function app, they'll see this log with every request. With Debug, we'll see it in our logs still.